pkg/scheduler/plugins/cpunormalization/plugin.go

```diff
@@ -0,0 +1,241 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	resourceapi "k8s.io/kubernetes/pkg/api/v1/resource"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config/validation"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	schedutil "k8s.io/kubernetes/pkg/scheduler/util"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/plugins/nodenumaresource"
+)
+
+const (
+	Name = "CPUNormalization"
+
+	preFilterStateKey = "PreFilter" + Name
+)
+
+var (
+	_ framework.EnqueueExtensions = &Plugin{}
+
+	_ framework.PreFilterPlugin = &Plugin{}
+	_ framework.FilterPlugin    = &Plugin{}
+	_ framework.ScorePlugin     = &Plugin{}
+)
+
+var nodeResourceStrategyTypeMap = map[config.ScoringStrategyType]scorer{
+	config.LeastAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                string(config.LeastAllocated),
+			scorer:              leastResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+	config.MostAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                string(config.MostAllocated),
+			scorer:              mostResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+}
+
+// Plugin checks if a node has sufficient CPU resources under CPU normalization.
+// It works like the NodeResourcesFit plugin but with CPU requests of CPUSet Pods amplified on Nodes with CPU normalization.
+type Plugin struct {
+	handle framework.Handle
+	resourceAllocationScorer
+}
+
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	pluginArgs, ok := args.(*config.NodeResourcesFitArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourcesFitArgs, got %T", args)
+	}
+	if err := validation.ValidateNodeResourcesFitArgs(nil, pluginArgs); err != nil {
+		return nil, err
+	}
+
+	if pluginArgs.ScoringStrategy == nil {
+		return nil, fmt.Errorf("scoring strategy not specified")
+	}
+
+	strategy := pluginArgs.ScoringStrategy.Type
+	scorePlugin, exists := nodeResourceStrategyTypeMap[strategy]
+	if !exists {
+		return nil, fmt.Errorf("scoring strategy %s is not supported", strategy)
+	}
+
+	return &Plugin{
+		handle:                   handle,
+		resourceAllocationScorer: *scorePlugin(pluginArgs),
+	}, nil
+}
+
+type preFilterState struct {
+	requestedMilliCPU int64
+	isCPUSet          bool
+}
+
+func (s *preFilterState) Clone() framework.StateData {
+	return s
+}
+
+func (p *Plugin) Name() string {
+	return Name
+}
+
+func (p *Plugin) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		{Resource: framework.Pod, ActionType: framework.Delete},
```

Comment:
IMO the plugin does not need to implement the interface since the native plugin such as `NodeResourcesFit` has already implemented it.